### ISSUE TITLE
argsort of masked array has wrong layout

### ISSUE THREAD

**Issue by agoose77** [scikit-hep/awkward#849]
Reproducer on Binder: [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gist/agoose77/0d2765f6d3b8e5b5b0cb75b28b7a95d0/HEAD?filepath=awkward-bug.ipynb)

Given an Awkward array and a mask:
```python 
label = ak.Array([[0, 1, 2, 3], [3, 3, 3, 2, 1]])
is_valid = label != 3
```
the result of `argsort(label.mask[is_valid])` is not a valid Array. 

It appears that the result does not maintain the mask, and so the `offsets` node in the layout is invalidated (it still refers to the masked structure).

```python
>>> label.mask[is_valid].layout
<ListOffsetArray64>
    <offsets><Index64 i="[0 4 9]" offset="0" length="3" at="0x0000026d8b10"/></offsets>
    <content><ByteMaskedArray valid_when="true">
        <mask><Index8 i="[1 1 1 0 0 0 0 1 1]" offset="0" length="9" at="0x0000026a7ad0"/></mask>
        <content><NumpyArray format="l" shape="9" data="0 1 2 3 3 3 3 2 1" at="0x0000026dab20"/></content>
    </ByteMaskedArray></content>
</ListOffsetArray64>
>>> ak.argsort(label.mask[is_valid]).layout
<ListOffsetArray64>
    <offsets><Index64 i="[0 4 9]" offset="0" length="3" at="0x0000026d8b10"/></offsets>
    <content><NumpyArray format="l" shape="5" data="0 1 2 1 0" at="0x0000028d0710"/></content>
</ListOffsetArray64>
```

I would expect the mask to persist after the call to `argsort` inasmuch as the resulting array has the same "shape" as the original array. 

### Awkward Array version
```python
>>> import awkward as ak
>>> ak.__version__
1.2.2
```

**Comment by jpivarski** [comment]
@ianna, the ByteMaskedArray node is somehow being lost. I don't think this has anything in particular to do with using the `.mask` mechanism; it's probably an issue with ByteMaskedArray as opposed to IndexedOptionArray.

**Comment by ianna** [comment]
@agoose77 - thanks for spotting the bug! https://github.com/scikit-hep/awkward-1.0/pull/850 should fix it.

**Comment by ianna** [comment]
fixed via #850
